packages/aws-cdk-lib/aws-ec2/lib/instance.ts

```diff
@@ -695,14 +695,24 @@ export class Instance extends Resource implements IInstance {
   }
 
   /**
-   * Apply CloudFormation update policies for the instance
+   * Apply CloudFormation update policies for the instance.
+   *
+   * If both initOptions.timeout and resourceSignalTimeout are specified,
+   * timeout is summed together, else timeout is set to the value of the specified field.
    */
   private applyUpdatePolicies(props: InstanceProps) {
+    if (props.initOptions?.timeout && props.resourceSignalTimeout) {
+      Annotations.of(this).addWarningV2('@aws-cdk/aws-ec2:setSetimeout', 'Both initOptions.timeout and resourceSignalTimeout fields are set, timeout is summed together. It is suggested that only one of the two fields is set');
+    }
+
     if (props.resourceSignalTimeout !== undefined) {
       this.instance.cfnOptions.creationPolicy = {
         ...this.instance.cfnOptions.creationPolicy,
         resourceSignal: {
-          timeout: props.resourceSignalTimeout && props.resourceSignalTimeout.toIsoString(),
+          ...this.instance.cfnOptions.creationPolicy?.resourceSignal,
+          timeout: props.initOptions?.timeout ?
+            props.resourceSignalTimeout.plus(props.initOptions.timeout).toIsoString() :
+            props.resourceSignalTimeout.toIsoString(),
```

Comment:
although it looks safe to add the 2 numbers, but this may cause a breaking change for customers if the sum of the 2 numbers is more than 12 hours .. this should be a very rare case, but ay be some customers set one of these numbers to the maximum value, and since we ignore the other one, it was working for them, but now it will stop working.

Also, I can see that the original intent was to sum these 2 numbers together based on this code https://github.com/aws/aws-cdk/blob/4fa7716e0bc50c06327bc30320c5871207a11c52/packages/aws-cdk-lib/aws-ec2/lib/instance.ts#L686-L694 That is why I think summing the 2 numbers is still Ok, but we should do it using a feature flag.

Reply:
one more thing it is not needed to do this change here, it is enough to change this snippet https://github.com/aws/aws-cdk/blob/4fa7716e0bc50c06327bc30320c5871207a11c52/packages/aws-cdk-lib/aws-ec2/lib/instance.ts#L592-L596 to move `this.applyUpdatePolicies(props);` before the if condition

Reply:
I think you're right that it may cause breaking changes when the sum is over the max threshold. Thanks for pointing that out.

For the second point, I disagree that we put it inside if statement because users could choose not to specify `props.init` and still setup the timeout value. Now if we move it inside, it won't happen anymore.

Reply:
i did not mean to put it inside the if condition, what I recommend is something like this 
```
this.applyUpdatePolicies(props); 

if (props.init) { 
   this.applyCloudFormationInit(props.init, props.initOptions); 
 } 
```

and to remove the addition changes that @michelle-wangg added to the `applyUpdatePolicies` method and depend that the addition logic is already exist in `applyCloudFormationInit` method some how:
https://github.com/aws/aws-cdk/blob/4fa7716e0bc50c06327bc30320c5871207a11c52/packages/aws-cdk-lib/aws-ec2/lib/instance.ts#L661-L695
See the login in `waitForResourceSignal`, by this change we do not need to duplicate the addition logic in 2 places.

For the warning message, I also prefer to move it to the constructor after the validations.

Reply:
Sounds good! I have updated the implementation to use a feature flag. Since I have not changed the logic `waitForResourceSignal`, even if `initOptions.Timeout` is not specified and `resourceSignalTimeout` is specified, the timeout will add the default 5 min to the resourceSignalTimeout.